dmenu.c

```diff
@@ -1504,6 +1659,8 @@ run(void)
 	int i;
 	#endif // PRESELECT_PATCH
 
+	XMapRaised(dpy, win);
+	grabfocus();
```

Comment:
Grabbing focus here makes it so you don't regain focus after closing dmenu, very disruptive behavior, it took a while for me to notice.